pkg/generators/k8s/ingress.go

```diff
@@ -0,0 +1,49 @@
+package k8s
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// -----------------------------------------------------------------------------
+// Public Functions - netv1.Ingress Helpers
+// -----------------------------------------------------------------------------
+
+// NewIngressForService provides a basic and opinionated *netv1.Ingress object for the provided *corev1.Service to expose it via an ingress controller for testing purposes.
+func NewIngressForService(ingressClass, path string, s *corev1.Service) *netv1.Ingress {
+	pathPrefix := netv1.PathTypePrefix
+	return &netv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: s.Name,
+			Annotations: map[string]string{
+				"kubernetes.io/ingress.class": ingressClass,
+				"konghq.com/strip-path":       "true",
```

Comment:
What was the background for setting this? The commit log indicates it was a fix, but doesn't say what wasn't working before.

The annotation set for an Ingress will need to vary per test. We can't test, for instance, different sets of plugins otherwise. We probably need to pass the annotation set in via an argument as such.

Reply:
Sounds good to me. I had considered this before but forgotten about it, seems :+1: to take care of now.

Reply:
a5a033c30d3ee355292cfa0b249a6328b6f7d5b3 resolves this :+1: